xarray/tests/__init__.py

```diff
@@ -95,6 +96,31 @@ def LooseVersion(vstring):
 
     dask.config.set(scheduler="single-threaded")
 
+
+class CountingScheduler:
+    """Simple dask scheduler counting the number of computes.
+
+    Reference: https://stackoverflow.com/questions/53289286/"""
+
+    def __init__(self, max_computes=0):
+        self.total_computes = 0
+        self.max_computes = max_computes
+
+    def __call__(self, dsk, keys, **kwargs):
+        self.total_computes += 1
+        if self.total_computes > self.max_computes:
+            raise RuntimeError(
+                "Too many computes. Total: %d > max: %d."
+                % (self.total_computes, self.max_computes)
+            )
+        return dask.get(dsk, keys, **kwargs)
+
+
+def raise_if_dask_computes(max_computes=0):
+    scheduler = CountingScheduler(max_computes)
+    return dask.config.set(scheduler=scheduler)
```

Comment:
this will probably fail since `dask` is not defined yet. I think you might have to add `import dask` to both `__call__` and at the beginning of this block

Reply:
`dask` is imported in `__init__.py`. However the `raise_if_dask_computes` context manager was also used in tests without dasks. E.g.:

https://github.com/pydata/xarray/blob/db4f03e467d13229512f8f7924dc142db1b9486b/xarray/tests/test_dataarray.py#L5136-L5137

Therefore I added a `dummy_context` - maybe there is a more elegant solution?